cmd/main.go

```diff
@@ -45,26 +51,33 @@ func main() {
 		command = reposerver.NewCommand()
 	case "argocd-cmp-server":
 		command = cmpserver.NewCommand()
-		isCLI = true
+		isArgocdCLI = true
 	case "argocd-dex":
 		command = dex.NewCommand()
 	case "argocd-notifications":
 		command = notification.NewCommand()
 	case "argocd-git-ask-pass":
 		command = gitaskpass.NewCommand()
-		isCLI = true
+		isArgocdCLI = true
 	case "argocd-applicationset-controller":
 		command = applicationset.NewCommand()
 	case "argocd-k8s-auth":
 		command = k8sauth.NewCommand()
-		isCLI = true
+		isArgocdCLI = true
 	default:
 		command = cli.NewCommand()
-		isCLI = true
+		isArgocdCLI = true
 	}
-	util.SetAutoMaxProcs(isCLI)
+	util.SetAutoMaxProcs(isArgocdCLI)
 
-	if err := command.Execute(); err != nil {
-		os.Exit(1)
+	// silence errors and usages since we'll be printing them manually.
+	// This is because if we execute a plugin, the initial
+	// errors and usage are always going to get printed that we don't want.
+	if isArgocdCLI {
+		command.SilenceErrors = true
+		command.SilenceUsage = true
 	}
+
+	err := command.Execute()
+	cli.HandleCommandExecutionError(err, isArgocdCLI, o)
```

Comment:
```suggestion
	cli.HandleCommandExecutionError(err, isArgocdCLI, o)
```
@blakepettersson @leoluz Please take a look at this now.

Reply:
This looks good (besides the fact that cobra doesn't provide a proper error type). Tks for the refactoring. I think we got in a much better state.